docs/sdks/python/knowledge-bases.mdx

```diff
@@ -0,0 +1,1069 @@
+    | 1   | Alice | 25  |
+    | 1   | Bob   | 30  |
+
+**Best practice**
+
+Ensure the `id_column` uniquely identifies each row to avoid unintentional data loss due to duplicate ID skipping.
+
+</Note>
+
+### Update Existing Data
+
+In order to update existing data in the knowledge base, insert data with the column ID that you want to update and the updated content.
+
+Here is an example of usage. A knowledge base stores the following data.
+
+```sql
++----------+-------------------+------------------------+
+| order_id | product           | notes                  |
++----------+-------------------+------------------------+
+| A1B      | Wireless Mouse    | Request color: black   |
+| 3XZ      | Bluetooth Speaker | Gift wrap requested    |
+| Q7P      | Laptop Stand      | Prefer aluminum finish |
++----------+-------------------+------------------------+
+```
+
+A user updated `Laptop Stand` to `Aluminum Laptop Stand`.
+
+```sql
++----------+-----------------------+------------------------+
+| order_id | product               | notes                  |
++----------+-----------------------+------------------------+
+| A1B      | Wireless Mouse        | Request color: black   |
+| 3XZ      | Bluetooth Speaker     | Gift wrap requested    |
+| Q7P      | Aluminum Laptop Stand | Prefer aluminum finish |
++----------+-----------------------+------------------------+
+```
+
+<Tip>
+Go to the *Complete Example* section below to find out how to access this sample data.
+</Tip>
+
+Here is how to propagate this change into the knowledge base.
+
+```python
+my_kb.insert_query(
+    server.databases.sample_data.tables.orders.filter(order_id='Q7P')
+)
+```
+
+The knowledge base matches the ID value to the existing one and updates the data if required.
+
+### Insert Data using Partitions
+
+TODO: os this available in python sdk?
+
+In order to optimize the performance of data insertion into the knowledge base, users can set up partitions and threads to insert batches of data in parallel. This also enables tracking the progress of data insertion process including cancelling and resuming it if required.
+
+Here is an example.
+
+```python
+project.query(
+    '''
+    INSERT INTO my_kb
+    SELECT order_id, product, notes
+    FROM sample_data.orders
+    USING
+        batch_size = 200,
+        track_column = order_id,
+        threads = 10,
+        error = 'skip';
+    '''
+)
+```
+
+The parameters include the following:
+
+* `batch_size` defines the number of rows fetched per iteration to optimize data extraction from the source. It defaults to 1000.
+
+* `threads` defines threads for running partitions. Note that if the [ML task queue](/setup/custom-config#overview-of-config-parameters) is enabled, threads are used automatically. The available values for include a number of threads to be used, for example, `threads = 10`, or a boolean value that defines whether to enable threads, setting `threads = true`, or disable threads, setting `threads = false`.
```

Comment:
The second sentence is a little bit confused to me, can we improve it?

Reply:
fixed